test/unit/SES/Transport/SESTransportTest.php

```diff
@@ -0,0 +1,228 @@
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('multiple `to` address');
+        $subject->send($message);
+    }
+
+    public function test_its_register_plugin_just_binds_to_event_listener()
+    {
+        $plugin  = new DummySwiftPlugin;
+        $subject = $this->newSubject();
+        $subject->registerPlugin($plugin);
+
+        $evt = $this->event_dispatcher->createSendEvent($subject, $this->getDummy(Swift_Mime_SimpleMessage::class));
+        $this->event_dispatcher->dispatchEvent($evt, 'sendPerformed');
+        $plugin->assertGotEvents([['sendPerformed', $evt]]);
+    }
+
+    public function test_its_send_method_sends()
+    {
+        $subject = $this->newSubject();
+        $message = $this->generateValidMessage();
+        $subject->send($message);
+        $this->assertSame([['RawMessage' => ['Data' => $message->toString()]]], $this->ses_client->getSentMails());
+    }
+
+    public function test_its_start_is_no_op()
+    {
+        $this->assertTrue($this->newSubject()->start());
+    }
+
+    public function test_its_stop_is_no_op()
+    {
+        $this->assertTrue($this->newSubject()->stop());
+    }
+
+    public function test_it_emits_timer_metric()
+    {
+        $subject = $this->newSubject();
+        $subject->send($this->generateValidMessage());
+        $this->metrics->assertCapturedOneTimer('ses-email-sent', 'ok');
+    }
+
+    public function test_it_emits_timer_metric_with_err_src_on_failure()
+    {
+        $this->ses_client = SpyingSESClient::sendRawEmailWillThrow();
+        $subject          = $this->newSubject();
+        try {
+            $subject->send($this->generateValidMessage());
+        } catch (\Exception $e) {
+        }
+        $this->metrics->assertCapturedOneTimer('ses-email-sent', 'err');
+    }
+
+    public function test_it_logs_success()
+    {
+        $subject = $this->newSubject();
+        $subject->send($this->generateValidMessage());
+        $this->assertSame(
+            [
+                [
+                    'level'   => LogLevel::INFO,
+                    'message' => 'Emailed foo@bar.com ()',
+                    'context' => [
+                        'aws_msg_id'     => NULL,
+                        'suppress_trace' => TRUE,
+                    ],
+                ],
+            ],
+            $this->log->records
+        );
+    }
+
+    public function test_it_logs_failure()
+    {
+        $this->ses_client = SpyingSESClient::sendRawEmailWillThrow();
+        $subject          = $this->newSubject();
+        try {
+            $subject->send($this->generateValidMessage());
+        } catch (\Exception $e) {
+        }
+        $this->assertSame(
+            LogLevel::ERROR,
+            $this->log->records[0]['level']
+        );
+        $this->assertSame(
+            'Failed to email `foo@bar.com` - [Exception] You asked me to throw',
+            $this->log->records[0]['message']
+        );
+    }
+
```

Comment:
Also worth a test that the internal exception is rethrown as EmailDeliveryFailedException and `$e->getPrevious()` is set correctly?